### PR TITLE
#22 store full text

### PR DESCRIPTION
**Pull request by albertocaparros**
A message from student when requesting previous content was needed finally. Right now it only appends the old content to the log if said log is empty (preventing other students connected to the session the get the whole content every time a new student connects). It'll probably be useful to have a student id or token in the future for things like this.

### CODE REVIEW

**Comment by brauliodiez** [comment]
Hi you can send a message only to the requester

```typescript
const handleErrorRoomIsBusy = (
  socketInfo: SocketInfo,
  connectionId: string
) => {
  const { socket } = socketInfo;
  console.log('**** ROOM BUSY ERROR *****');
  socket.emit('error-app', ErrorCodes.roomBusy);
};

```

**Comment by albertocaparros** [comment]
Good call, I reused the emitting code so it didn't cross my mind. Refactored!

**Comment by albertocaparros** [comment]
Refactored what you mentioned yesterday. Socket message clarity + storing every append text instead of individual socket call

File: back/src/messages/input-processor.ts [hunk 1]
```diff
@@ -47,11 +48,23 @@ export const processInputMessage = (
...
+      );
+      break;
   }
 
   return outputActionCollection;
 };
 
+const handleGetStudentContent = (socketInfo: SocketInfo, room: string) => {
+  return [
+    { type: OutputMessageTypes.STUDENT_SEND_FULL_CONTENT, payload: room },
```
Comment (brauliodiez) [inline back/src/messages/input-processor.ts:64]: We don't need to pass the room, we can obtain it from the socket info

File: back/src/messages/input-processor.ts [hunk 2]
```diff
@@ -47,11 +48,23 @@ export const processInputMessage = (
...
+        socketInfo,
+        action.payload
+      );
+      break;
   }
 
   return outputActionCollection;
 };
 
+const handleGetStudentContent = (socketInfo: SocketInfo, room: string) => {
```
Comment (brauliodiez) [inline back/src/messages/input-processor.ts:62]: This should be named, handleRequestGetStudentContent

File: back/src/messages/output-processor.ts [hunk 3]
```diff
@@ -37,9 +41,20 @@ export const processOuputMessage = (socketInfo: SocketInfo, action: Action) => {
       handleAppendText(socketInfo, action.payload);
     case OutputMessageTypes.REPLACE_FULL_TEXT:
       handleReplaceFullText(socketInfo, action.payload);
+    case OutputMessageTypes.STUDENT_SEND_FULL_CONTENT:
+      handleStudentSendContent(socketInfo, action.payload);
   }
 };
 
+const handleStudentSendContent = (socketInfo: SocketInfo, room: string) => {
```
Comment (brauliodiez) [inline back/src/messages/output-processor.ts:49]: I thin we can remove room from there, and obtain it via socketInfo, then we get the content or something like that

File: back/src/storage/session.ts [hunk 4]
```diff
@@ -31,6 +38,40 @@ export const addNewUser = (
...
+      roomCollectionSession,
+      (draftState: RoomInfo[]) => {
+        draftState.push(newRoomInfo);
+      }
+    );
+    roomCollectionSession = nextRoomCollectionSession;
+  }
+
+  console.log('roomcollectionsession', roomCollectionSession);
+};
```
Comment (brauliodiez) [inline back/src/storage/session.ts:65]: Let's apply some clean code here (same level of abstraction principle), something like

```tsx
export const saveRoomInfo = (newRoomInfo: RoomInfo) => {
  const roomIndex = getRoomIndexByName(newRoomInfo.room);

  roomCollectionSession =  (roomIndex === -1)  ?
                                  updateRoomContent(index, content)
                                 :
                                 insertRoom(newRoomInfo); 
}
```

The create the detail functions.

By following this code is easier to follow

File: front/src/core/const.ts [hunk 5]
```diff
@@ -9,12 +9,15 @@ export const SocketOuputMessageLiteral = {
 
 export const SocketEmitMessageTypes = {
   TRAINER_APPEND_TEXT: 'TRAINER_APPEND_TEXT', // Master creates a user story
+  TRAINER_SAVE_CONTENT: 'TRAINER_SAVE_CONTENT', // Master saves current content
```
Comment (brauliodiez) [inline front/src/core/const.ts:12]: Not sure what this message is for

File: front/src/pods/student/student.container.tsx [hunk 6]
```diff
@@ -27,22 +29,31 @@ export const PlayerContainer = () => {
...
-
       if (msg.type) {
         const { type, payload } = msg;
 
         switch (type) {
           case SocketReceiveMessageTypes.APPEND_TEXT:
             appendToLog(payload);
             break;
+          case SocketReceiveMessageTypes.STUDENT_GET_FULL_CONTENT:
+            appendToLog(payload);
```
Comment (brauliodiez) [inline front/src/pods/student/student.container.tsx:40]: If we get the full content we should replace, not append

File: front/src/core/const.ts (hunk 5 above)
Comment (albertocaparros) [reply front/src/core/const.ts:12]: Residue from previous iteration (before saving content on append). Removing.

File: front/src/pods/student/student.container.tsx (hunk 6 above)
Comment (albertocaparros) [reply front/src/pods/student/student.container.tsx:40]: No replace or set function on use-log.hook, so I was unsure if it was deliberate or it wasn't needed until now. New setLog functionality created.

**Comment by brauliodiez** [comment]
It doesn't work, I think you have forgotten to list to the GET_FULL_CONTENT Message on student container:

```tsx
localSocket.on(SocketOuputMessageLiteral.MESSAGE, msg => {
      console.log(msg);

      if (msg.type) {
        const { type, payload } = msg;

        switch (type) {
          case SocketReceiveMessageTypes.APPEND_TEXT:
            appendToLog(payload);
            break;

          // TODO Wait for SEND_FULL_CONTENT Message
          // setLog(payload);
        }
      }
    });
  };
```

**Pull request merged**